components/camel-xchange/src/main/java/org/apache/camel/component/xchange/XChangeComponent.java

```diff
@@ -36,15 +38,29 @@ protected Endpoint createEndpoint(String uri, String remaining, Map<String, Obje
         // Set the the required name of the exchange
         configuration.setName(remaining);
 
-        // Get the XChange implementation
-        Class<? extends Exchange> exchangeClass = configuration.getXChangeClass();
-        Assert.notNull(exchangeClass, "XChange not supported: " + configuration.getName());
-        
-        // Create the XChange and associated Endpoint
-        XChange exchange = new XChange(ExchangeFactory.INSTANCE.createExchange(exchangeClass));
+        XChange exchange = createXChange(configuration);
         XChangeEndpoint endpoint = new XChangeEndpoint(uri, this, configuration, exchange);
         
         return endpoint;
     }
 
+    public XChange getXChange() {
+        return exchange;
+    }
+    
+    private synchronized XChange createXChange(XChangeConfiguration configuration) {
```

Comment:
Its better to do this in doStart so you dont need synchronized code blocks

Reply:
I'd like to create the XChange once so it can be shared among all endpoints. For this I'd need some piece of information that names the exchange. This is currently encoded in the URI and (if IIUC) becomes available when the endpoint is created.

If I do this initialization in Endpoint.doStart(), I'd still need to synchronize unless it is guaranteed that endpoints are not started in parallel (also in the future).

If I do this initialization in Component.doStart(), I'd not know how to get at the piece of information that names the exchange.